### ISSUE TITLE
Include theme settings in saveViewState/restoreViewState

### ISSUE THREAD

**Issue by mbarkhau** [Microsoft/monaco-editor#526]
monaco-editor version: 0.9.0
Browser: Any
OS: Any

There is the command "Toggle High Contrast Theme" which does not appear to be stored/restored by the *ViewState functions. Calls to `monaco.editor.setTheme` also don't seem to be preserved, but maybe that's the same thing.

I guess this might be because the themes are not an editor specific, but rather a global setting?

**Comment by alexandrudima** [comment]
Yes, themes are a global setting.

Mostly, the theme is controlled by the integrator via `monaco.editor.setTheme()`. 

There is one exception to that, which you've already found, which is the command "Toggle High Contrast Theme". The command is meant to be used by end-users if they are having trouble with reading the content inside an editor. The command calls `monaco.editor.setTheme()` in its implementation. e.g. an integrator decides to customize the editor to use colors which don't meet the 4:5:1 contrast ratio bar. The only recourse a visually impaired end user has is to run the "Toggle High Contrast Theme" command.

TL;DR: I wouldn't worry too much about this command, I am expecting only users which are really needing it would run it.

**Comment by mbarkhau** [comment]
My issue is with the `*ViewState` functions (`saveViewState` and `restoreViewState`). I'd like to restore the view state of the user when they resume a session, but currently the theme is not saved/restored by these calls, so I would have to do that separately.

**Comment by alexandrudima** [comment]
oh, I see. You'd like to know if the user switched the editor to the high contrast theme via this command.

Well, you can always build an experience / settings in your app to allow for users to switch to high contrast your entire app / website, not just the editor.

The "Toggle High Contrast Theme" command is more like an emergency command which any user can invoke if editor hosts haven't considered high contrast.

But if you are considering supporting high contrast, I suppose your entire application needs to support it, so you can build a separate experience for that, and simply set us to high contrast via `monaco.editor.setTheme('hc-black')` for the users which chose High Contrast in your app's settings.

**Comment by mbarkhau** [comment]
Really this has practically nothing to do with the high contrast mode. As the title of the ticket says, it's about the `saveViewState/restoreViewState` functions, which don't save/restore the theme settings.

**Comment by mbarkhau** [comment]
Maybe this is a misunderstanding on my part, are the theme settings not considered part of the view state?

**Comment by alexandrudima** [comment]
Yes, `editor.saveViewState` stores:
 * cursor position
 * scroll location
 * folded sections
for a certain model when it is connected to an editor instance.

Once the same model is connected to the same or a different editor instance, `editor.restoreViewState` can be used to restore the above listed state.

There are very many things that influence how rendering occurs:
 * the current theme
 * the current wrapping settings set on the editor
 * the enablement of a minimap, etc.
 * the current language configured for a model
 * etc.

But these two methods are about reusing the same instance of an editor with multiple intances of a model and being able to serialize that state via `JSON.stringify` in a persistent location.

**Comment by mbarkhau** [comment]
Ok then, i guess i will have to store the theme info separately. Thank you.
